libs/blocks/table/table.js

```diff
@@ -153,6 +153,32 @@ function handleAddOnContent(table) {
   table.addEventListener('mas:resolved', debounce(() => { handleEqualHeight(table, '.row-heading'); }));
 }
 
+function setTooltipPosition(el) {
+  const tooltips = el.querySelectorAll('.milo-tooltip');
+
+  tooltips.forEach((tooltip) => {
+    const classesToCheck = ['top', 'bottom', 'left', 'right'];
+    const defaultClass = classesToCheck.find((className) => tooltip.classList.contains(className)) || 'right';
+    tooltip.setAttribute('data-tooltip-default-position', defaultClass);
+
+    tooltip.addEventListener('mouseenter', () => {
+      const isTabletOrMobile = ['TABLET', 'MOBILE'].includes(defineDeviceByScreenSize());
+      const isRtl = document.documentElement.dir === 'rtl';
+
+      if (!((el.getBoundingClientRect().width === window.innerWidth) || isTabletOrMobile)) {
```

Comment:
Are we actually concerned with the width of the element? Or would it be its left/right boundaries and making sure they don't exceed the viewport? If it's the former, couldn't we use `el.clientWidth` or `el.scrollWidth`? `getBoundingClientRect` does create a bit of a performance overhead and if it's easy enough to avoid, we should. If it's the latter, we should update the logic to check for left/right boundaries.